SpatialGDK/Documentation/content/spatialos-concepts/schema.md

```diff
@@ -1,31 +1,56 @@
 
 Schema is defined in `.schema` files and written in schemalang.  When you use the GDK, the schema files and their contents are generated and deleted automatically so you do not have to write or edit schema files manually. The GDK generates and deletes schema for you, when you start schema generation.
 
 #### How to generate schema
 
-To generate schema, select the **Schema** button in the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes with replicated properties to generate the required schema files and then updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).
+There are two ways to generate schema for your project:
 
-![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/schema-button.png)
+* **Generate schema (Full Scan)** 
 
-_Image: In the GDK toolbar in the Unreal Editor, select **Schema**_
+    To generate schema for all classes* in your project:<br/>
+
+    In the Unreal Editor, on the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons), open the **Schema** drop-down menu and select **Schema (Full Scan)**.<br/> You must select **Schema (Full Scan)** the first time you generate schema for a project. 
+    <br/> ![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/schema-button-full-scan.png)<br/>
+    _Image: In the GDK toolbar in the Unreal Editor, select **Schema (Full Scan)**_<br/>
+    <br/>When you select **Schema (Full Scan)**, the GDK automatically iterates through all classes* in your project to generate the schema files, and then updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase). <br/>
+
+    Run a full scan the first time you generate schema for your project, and whenever you need to generate schema for classes* that are not currently loaded by the Editor.<br/>For example: You need to select **Schema (Full Scan)** if you didn't generate schema after adding a Blueprint to a level, and that level is no longer open in the Editor.<br/><br/>
+
+* **Generate Schema (Iterative)**
+
+    To generate schema for classes* that are currently loaded by the Editor: <br/>
+
+    Select  **Schema** in the [GDK Toolbar]({{urlRoot}}/content/toolbars#buttons). The GDK automatically iterates through classes* that are currently loaded by the Editor, generates the schema files and updates the [SchemaDatabase]({{urlRoot}}/content/glossary#schemadatabase).<br/>
+    ![Toolbar]({{assetRoot}}assets/screen-grabs/toolbar/schema-button.png)<br/>
+    _Image: In the GDK toolbar in the Unreal Editor, select **Schema**_<br/><br/>
+
+    You must run an iterative scan when you add or change any classes* that you want to deploy to SpatialOS.
```

Comment:
```suggestion
    You must run an iterative scan when you add or change replicated properties in any classes* that you want to deploy to SpatialOS.
```

Reply:
Do we need this sentence? Perhaps just refer to the *When to generate schema* section

Reply:
Nah, probably not. Good shout.